src/ontology/hp.Makefile

```diff
@@ -636,3 +643,27 @@ $(TMPDIR)/hp-%-merged.owl: hp-base.owl tmp/%.owl
 
 mappings: 
 	$(MAKE_FAST) ../mappings/hp-snomed.lexmatch.sssom.tsv
+
+###########################################
+### Testing new custom plugin system ######
+###########################################
+
+# Make sure ROBOT knows where to find plugins
+export ROBOT_PLUGINS_DIRECTORY=$(TMPDIR)/plugins
+
+# List of "repository plugins" (plugins permanently installed in top-level plugins directory, entirely managed by users)
+ROBOT_REPOSITORY_PLUGINS=$(wildcard ../../plugins/*.jar)
+
+# Standard rule to create symlinks in $(TMPDIR)/plugins, so that the repository plugins
+# are visible to ROBOT
+prepare-robot-plugins:
+	for plugin in $(ROBOT_REPOSITORY_PLUGINS) ; do ln $plugin $(TMPDIR)/plugins ; done
```

Comment:
This does not throw an error when the ../../plugins directory does not exist, which is good.

Reply:
But this _will_ throw an error if there are some plugins in the `../../plugins` directory but the destination directory (`$(TMPDIR)/plugins`) does not exist. We need a `mkdir -p $(TMPDIR)/plugins` before attempting to create the symlinks.

Reply:
No. We’re creating a _normal_ link here, not a _symbolic_ one (or a “hard” link instead of a “soft” one) – my mistake in the previous comment for calling them “symlinks” out of habit.

If the original file in `../../plugins/` is deleted, the newly added link in `$(TMPDIR)/plugins` will still be there, and will still point to the same contents (technically the same _inode_) on the disk.

(FYI: The reason we are creating normal links instead of symbolic links is that for some reasons, from within the ODK ROBOT would not be able to correctly follow the symlinks, even if both the symlinks and the files they are pointing to are both visible from the container – I’ve tried.)

Reply:
I’ll be curious to see whether this works as expected on Windows, though. AFAIK Windows has no such concepts as _links_ and _inodes_, so I don’t know how Docker on Windows will handle that…